fix: persist Agent Executor title after save/reload

aryash45 opened the pull request:
- Save agent name + version into `hardcodedValues.agent_name`
- Display `agent_name` on load (fallback to live agent)
- Uses existing backend schema field
- Fixes #11041

### Changes
- Modified `CustomNode.tsx` to:
  - Save computed title (`agent.name vX.Y`) into `hardcodedValues.agent_name` when agent is selected
  - Display `agent_name` first on load, with fallback to live agent data
  - Use existing `agent_name` field in backend schema — **no backend changes required**
- Added `react-use` for `useDebounce` (already installed via `--legacy-peer-deps`)

### Why This Change Is Needed
When an **Agent Executor** block is added and an agent with a version is selected (e.g., `Researcher v2.1`), the title updates correctly in the UI.  
However, after **saving the project and reloading**, the title reverts to the generic `"Agent Executor"`.

This happens because the title is computed at render time but **never persisted** in the block’s saved data. This PR fixes it by using the **already-existing `agent_name` field** in the backend schema to store and restore the full title.

---

### Checklist
#### For code changes:
- [x] I have clearly listed my changes in the PR description
- [x] I have made a test plan
- [ ] I have tested my changes according to the test plan:
  <!-- Could not test locally due to environment issues (Docker/npm conflicts) -->
  > **Test Plan (for maintainers):**
  > - [ ] Add **Agent Executor** block to a new project
  > - [ ] Select an agent with a version (e.g., `Researcher v2.1`)
  > - [ ] Confirm block title shows `Researcher v2.1`
  > - [ ] **Save** the project
  > - [ ] **Refresh** the page
  > - [ ] Confirm title **still shows** `Researcher v2.1`
  > - [ ] (Optional) Edit title manually → save → refresh → should persist

#### For configuration changes:
- [x] `.env.default` is updated or already compatible with my changes
- [x] `docker-compose.yml` is updated or already compatible with my changes
- [x] I have included a list of my configuration changes in the PR description (under **Changes**)

---

**Note**: Could not run full local environment due to dependency conflicts (`tailwindcss` version mismatch).  
However, the fix uses **only existing schema fields** and follows the **exact pattern** used elsewhere in the codebase.

Ready for CI & maintainer testing!

aryash45 committed:
fix: persist Agent Executor title after save/reload

- Save agent name + version into hardcodedValues.agent_name
- Display agent_name on load (fallback to live agent)
- Uses existing backend schema field
- Fixes #11041

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Significant-Gravitas/AutoGPT?pullRequest=11255) <br/>All committers have signed the CLA.

aryash45 committed:
Merge branch 'dev' into fix/11041-persist-agent-title

aryash45 commented:
hey @Pwuts please review and let me know if needed any changes

aryash45 committed:
fix: properly display agent title with version in CustomNode title rendering

aryash45 committed:
Merge branch 'dev' into fix/11041-persist-agent-title

aryash45 committed:
Merge branch 'dev' into fix/11041-persist-agent-title

aryash45 committed:
Merge branch 'dev' into fix/11041-persist-agent-title

AutoGPT-Agent commented:
Thanks for this PR to fix the Agent Executor title persistence issue!

### Feedback

1. Your implementation approach looks good - using the existing `agent_name` field in the backend schema is a clean solution.

2. There appears to be a merge conflict in the `package.json` file - there are duplicate entries for `@chromatic-com/storybook` with different versions (4.1.1 and 4.1.2). This will need to be resolved before merging.

3. While I understand you couldn't test locally due to environment issues, it would be good to get confirmation from maintainers that they've tested according to your test plan before final approval.

Overall, this looks like a solid fix that addresses the issue in a clean way. Just need to resolve that package.json conflict, and we should be good to go!

Abhi1992002 reviewed (changes requested):
Thank you @aryash45 for contributing! 😊 I have requested some changes. I will approve it if you fix or answer those questions.

I have also tested this PR locally, and it’s not working. I think it’s because you are not persisting the agent details in hardcodedValues. I believe you forgot to call a function inside a side effect (I have already mentioned that). 

I think you can also do something like this — it would be easier:
- On initial render, check if metadata has a `customTitle` or not. If not, then save the agent name in metadata and then save it. Now you have persisted the agent name in the custom title, and everything will work fine. 

It’s a temporary solution, or the solution you have given is also temporary. For a permanent one, you can see why the title is changing on saving. Are we sending some wrong title to the backend when saving? [You need to go through the saving logic] 
Maybe you will find some issue.

Abhi1992002 commented on autogpt_platform/frontend/src/app/(platform)/build/components/legacy-builder/CustomNode/CustomNode.tsx:
```diff
@@ -466,6 +511,20 @@ export const CustomNode = React.memo(
+      }
+      setLocalAgent({id:agent.id, name:agentName, version:version});
+      setHardcodedValues(newHardcodedValues);
+    },[data.hardcodedValues, setHardcodedValues]);
```
That’s why it doesn’t work locally. You haven’t persisted agent details in hardcoded values.

aryash45 committed:
fix: address reviewer comments on PR #11255

- Remove unused react-use dependency and useDebounce import
- Remove localAgent state (not needed; agent info lives in hardcodedValues)
- Fix illegal setState calls during render phase (setLocalAgent in if-block)
- Simplify agentTitle to read directly from hardcodedValues.agent_name
- Clean up updateAgentInBlock: remove unused title var and localAgent ref
- Fix React.memo comparator: remove unnecessary id check per reviewer

aryash45 committed:
fix: wire up updateAgentInBlock to actually persist agent name

When an AgentExecutor block loads with a graph_id but no agent_name,
fetch the graph name via BackendAPI and call updateAgentInBlock
to persist it into hardcodedValues.

Previously updateAgentInBlock was defined but never called, meaning
agent_name/agent_version were never set and agentTitle always
fell back to 'Agent Executor' after save/reload.

aryash45 reviewed (commented):
done
